### PR TITLE
Clean up --help

### PR DESCRIPTION
**Pull request by grouma**
Clean up `--help` to better align with other commands captured in Dart CLI.

Closes https://github.com/dart-lang/test/issues/1330

### CODE REVIEW

File: pkgs/test_core/lib/src/runner/configuration/args.dart [hunk 1]
```diff
@@ -27,13 +27,13 @@ final ArgParser _parser = (() {
   parser.addFlag('help',
       abbr: 'h', negatable: false, help: 'Shows this usage information.');
```
Comment (natebosch) [inline pkgs/test_core/lib/src/runner/configuration/args.dart:28]: Should this be `Show` for consistency?

```suggestion
      abbr: 'h', negatable: false, help: 'Show this usage information.');
```

File: pkgs/test_core/lib/src/runner/configuration/args.dart [hunk 2]
```diff
@@ -118,25 +119,24 @@ final ArgParser _parser = (() {
...
   parser.addOption('file-reporter',
-      help: 'The reporter used to write test results to a file.\n'
+      help: 'Set the reporter used to write test results to a file.\n'
           'Should be in the form <reporter>:<filepath>, '
-          'e.g. "json:reports/tests.json"');
+          'Example: "json:reports/tests.json"');
   parser.addFlag('verbose-trace',
-      negatable: false,
-      help: 'Whether to emit stack traces with core library frames.');
+      negatable: false, help: 'Emit stack traces with core library frames.');
```
Comment (natebosch) [inline pkgs/test_core/lib/src/runner/configuration/args.dart:134]: Do you know why this isn't negatable? This came up on the doc.

Comment (grouma) [reply pkgs/test_core/lib/src/runner/configuration/args.dart:134]: I found the PR that added the flag: https://github.com/dart-lang/test/commit/b61a5362bb37eb7a9ffa6ccb2f7837d79701b4c6

Doesn't have a reason though.

File: pkgs/test/pubspec.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 name: test
-version: 1.16.0-nullsafety.2
+version: 1.16.0-nullsafety.3
```
Comment (devoncarew) [inline pkgs/test/pubspec.yaml:2]: I assume this means that users won't see the help text changes until they're using a null safe version of package:test?

Comment (grouma) [reply pkgs/test/pubspec.yaml:2]: That I don't know. @natebosch

Comment (natebosch) [reply pkgs/test/pubspec.yaml:2]: That is correct, but using a null safe compatible version of `package:test` does not require that the user migrates to null safety.

**Pull request merged**